### PR TITLE
Merge in main for stable release being tested in Test, then deployed in production

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: terraform/run-init-on-aws.py [hunk 1]
```diff
@@ -597,6 +598,11 @@ def main():
         initializer.tf_cmd = "terragrunt"
     if args.working_dir:
         working_dir = args.working_dir.resolve()
+    elif args.terragrunt:
+        working_dir = Path.cwd()
```
Comment (nutjob4life) [inline terraform/run-init-on-aws.py]: Is this assumption okay? If a user specifies `--terragrunt`, the caller's $CWD becomes the project root. Makes it "location dependent" and replaces the initializer's default paths (and maybe worsened if called from a wrapper or CI).

It might be better to require `--working-dir` when in Terragrunt mode, or derive the repository root deterministically.

File: terraform/run-init-on-aws.py [hunk 2]
```diff
@@ -61,9 +61,10 @@ def print_section(self, emoji: str, message: str):
 
     def check_terraform_directory(self) -> bool:
         """Check that the opensearch_serverless and applications Terraform state directories exist."""
+        marker = "terragrunt.hcl" if self.tf_cmd == "terragrunt" else "main.tf"
```
Comment (nutjob4life) [inline terraform/run-init-on-aws.py:66]: This is a little brittle; `self.tf_cmd` could be a full path like `/usr/local/bin/terragrunt` or on Windows `terragrunt.exe`. Maybe a separate boolean like `use_terragrunt` could be sturdier.

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Got a couple concerns on `run-init-on-aws.py`; see the interspersed comments. But I feel like we're close.

File: terraform/run-init-on-aws.py (hunk 1 above)
Comment (tloubrieu-jpl) [reply terraform/run-init-on-aws.py]: AI came up with that logic and I only verified that it worked in my case ! I tried to clean the logic a little bit now, I believe that looks better.

File: terraform/run-init-on-aws.py (hunk 2 above)
Comment (tloubrieu-jpl) [reply terraform/run-init-on-aws.py:66]: The value self.tf_cmd is assigned within the script and the only possible values are terraform and terragrunt, although it is not clearly stated in the code.
That is also a limitation of the script to only run when terragrunt or terraform are in the PATH of the user and that script would not work on Window and we don't need it to for now at least.

I added these details in the inline help of the tool.

The script is used to initialize and test a new registry, so the pool of users targeted is quite limited, would be me and next full automation ! But thanks for the feedback, the script has fully be generated with AI and I lately missed to keep an eye on the code.

**Review by jordanpadams — APPROVED** [review]
